main.tf

```diff
@@ -51,18 +51,14 @@ resource "aws_ecs_cluster" "this" {
 ################################################################################
 
 locals {
-  # We are merging these together so that we can reference the ECS capacity provider
-  # (ec2 autoscaling) created in this module below. Fargate is easy since its just
-  # static values, but the autoscaling cappacity provider needs to be self-referenced from
-  # within this module
-  cluster_capacity_providers = merge(
-    var.fargate_capacity_providers,
-    { for k, v in var.autoscaling_capacity_providers : k => merge(aws_ecs_capacity_provider.this[k], v) }
+  default_capacity_providers = merge(
+    { for k, v in var.fargate_capacity_providers : k => v if var.default_capacity_provider_use_fargate },
+    { for k, v in var.autoscaling_capacity_providers : k => v if !var.default_capacity_provider_use_fargate }
   )
 }
 
 resource "aws_ecs_cluster_capacity_providers" "this" {
-  count = var.create ? 1 : 0
+  count = var.create && length(merge(var.fargate_capacity_providers, var.autoscaling_capacity_providers)) > 0 ? 1 : 0
```

Comment:
This count will probably fail when user creates the capacity provider resource outside of the module and passes it as variable to the module, because of the famous error - computed values can't be used in count.

Solution can be to rewrite to for_each or introduce another variable?

Reply:
hmm, I thought because this was just an `on/off` flag and not iterating over it, I would be safe. But thats a good point regardless - let me add external providers to the `complete` example to validate and increase test coverage

Reply:
actually, we already have this and it works as intended. The EC2/Autoscaling groups are created externally through our `terraform-aws-autoscaling` module and passed in since we do not create the ASG(s) in this module. I think because this is just doing a simple length check, that we don't encounter any of the unknown values issues